lsp/nls/src/cache.rs

```diff
@@ -40,20 +40,33 @@ impl CacheExt for Cache {
         initial_env: &Environment,
         lin_cache: &mut HashMap<FileId, Completed>,
     ) -> Result<CacheOp<()>, CacheError<TypecheckError>> {
-        if !self.terms_mut().contains_key(&file_id) {
+        if !self.terms().contains_key(&file_id) {
             return Err(CacheError::NotParsed);
         }
 
+        if let Ok(CacheOp::Done(ids)) = self.resolve_imports(file_id) {
+            for id in ids {
+                if id == file_id {
```

Comment:
To be checked, but I'm believe `resolve_imports` doesn't return the `file_id` given as a parameter, but just the ones of the import encountered inside `file_id`?

Reply:
Yes, it goes arbitrarily deep and returns all the file ids of the imports it can reach.

Reply:
yes, but `transform::resolve_import` or whatever is called only return the import that has not been resolved before I think. It's not very important but the cache should already be resilient to cyclic imports (I think there are even tests for that), so this is most probably not needed, in that `file_id` shouldn't be in the return value of `resolve_imports(file_id)`